tests/FSharp.Stats.Tests/Distributions.fs

```diff
@@ -29,30 +29,314 @@ let distanceFunctionsTests =
+                Expect.throws (fun () -> testCase.Variance |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.StandardDeviation |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 0. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 10. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF -infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF -1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF nan |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 0. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 10. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF -infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF -1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF nan |> ignore) "Should throw"
+            testCase "chiSquaredNan" <| fun () ->
+                let testCase = Continuous.chiSquared nan
+                Expect.throws (fun () -> testCase.Mean |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.Variance |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.StandardDeviation |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 0. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF 10. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF -infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF -1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.CDF nan |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 0. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF 10. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF -infinity |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF -1. |> ignore) "Should throw"
+                Expect.throws (fun () -> testCase.PDF nan |> ignore) "Should throw"
```

Comment:
Added them for 2 reasons:
1. Because of the possibility that changes _could_ theoretically be done on those methods instead of the functions they call
2. To cover those lines in the codecoverage report

Reply:
You are right, I missed that one. But anyway, to save resources one test per method and case would be enough I think. No matter what you input in `(chiSquared -1.).CDF x` will fail because of the negative value rather than what x is, so you could just check it once. Or do I miss something?

```f#
let testCase = Continuous.chiSquared -1.
Expect.throws (fun () -> testCase.Mean |> ignore) "Should throw" //<- messages could be more descriptive "chiSquared distribution cannot be initialized with negative degrees of freedom"
Expect.throws (fun () -> testCase.Variance |> ignore) "Should throw"
Expect.throws (fun () -> testCase.StandardDeviation |> ignore) "Should throw"
//Expect.throws (fun () -> testCase.CDF 0. |> ignore) "Should throw"
Expect.throws (fun () -> testCase.CDF 1. |> ignore) "Should throw"
//Expect.throws (fun () -> testCase.CDF 10. |> ignore) "Should throw"
//Expect.throws (fun () -> testCase.CDF infinity |> ignore) "Should throw"
//Expect.throws (fun () -> testCase.CDF -infinity |> ignore) "Should throw" 
//Expect.throws (fun () -> testCase.CDF -1. |> ignore) "Should throw"
...
```

Reply:
> You are right, I missed that one. But anyway, to save resources one test per method and case would be enough I think. No matter what you input in `(chiSquared -1.).CDF x` will fail because of the negative value rather than what x is, so you could just check it once. Or do I miss something?

You are right, but what about if someone would put in a whole new algorithm into the method? Pretty sure this will never occur but I treated tests for also tackling sheer impossible evolutions.
I mean, I can delete most of the Expecto tests you exemplarily showed if it's better performance-wise.